### DISCUSSION TITLE
Battery status polling when charger unplugged

### DISCUSSION THREAD

**Discussion by vonguyenkhang** [VR-25/acc#121] [Q&A] [answered]
Hi, when charger is unplugged, I see that acc log still updates every 10sec. Does it have any effect on battery life? (e.g. CPU is constantly woke up and unable to enter deep sleep)

**Comment by VR-25** [comment, marked as answer]
The daemon is designed with efficiency in mind.
There were attempts to make it even more efficient by taking advantage of interrupts/events. Unfortunately, only a handful of Android systems offer that possibility.
Still, there's virtually no negative impact on battery life.

**Reply by vonguyenkhang** [reply]
I've tried stopping accd when the charger is unplugged and restarting accd when the charger is plugged in, but after monitoring for one week I can't tell if there's any difference in battery life at all. 
So, as you said, probably it's better to make the daemon simple and efficient than to add extra code for switching between interrupts/events and polling for different ROMs.
